src/string.cpp

```diff
@@ -411,6 +411,25 @@ gb_internal String concatenate4_strings(gbAllocator a, String const &x, String c
 	return make_string(data, len);
 }
 
+#if defined(GB_SYSTEM_WINDOWS)
+gb_internal String escape_char(gbAllocator a, String s, char cte) {
+	isize buf_len = s.len;
+	u8 *buf = gb_alloc_array(a, u8, buf_len);
```

Comment:
The newly allocated String has the same length as input, even though zero or more backslashes will be added. This will result in an out of bounds access if something is actually escaped.

Reply:
Not very familiar with C++.
Would something like work here
```
isize cte_count = 0;
for (isize j = 0; j < s.len; j++)
	if (s.text[j] == cte)
		cte_count++;
u8 *buf = gb_alloc_array(a, u8, buf_len+cte_count);
```

Reply:
Yes

Reply:
changed accordingly